projects/map-ol/src/lib/map-ol.component.ts

```diff
@@ -444,6 +474,12 @@ export class MapOlComponent implements OnInit, AfterViewInit, AfterViewChecked,
       }
 
       if (tempControls.length) {
+        /**
+         * check tempControls dose not include oldControls
+         * https://medium.com/@alvaro.saburido/set-theory-for-arrays-in-es6-eb2f20a61848#f22b
+         */
+        const difference = oldControls.filter(x => !tempControls.includes(x));
```

Comment:
All the controls in `tempControls` are new instances of the Control-class, right?
So `difference` is always identical to `oldControls`, right?

Same as in this example: 
```js
let arr = [{"a": 1}, {"b": 2}]
undefined
let el = {"a": 1}
undefined
arr.includes(el)
false
```

Reply:
The oldControls are Controls added with the MapService in a route on ngOnInit.
At this time map-component is not already created, so we have to collect them and add them to the controls inserted by the Controls Input of the map which are pushed in tempControls.
Normally there should not by any duplicates in map.getControls() before, but to filter them out if there are any, i tried to get the difference.